code/modules/mob/dead/observer/observer.dm

```diff
@@ -85,10 +85,15 @@ GLOBAL_VAR_INIT(observer_default_invisibility, INVISIBILITY_OBSERVER)
 		name = capitalize(pick(GLOB.first_names_male)) + " " + capitalize(pick(GLOB.last_names))
 	real_name = name
 
+	..()
+
+/mob/dead/observer/Initialize(mapload, flags=1)
+	. = ..()
+	var/mob/body = src.mind?.current
 	//starts ghosts off with all HUDs.
-	toggle_all_huds_on(body)
+	if(istype(body))
+		toggle_all_huds_on(body)
```

Comment:
That was my first thought as well, and, yeah, that didn't work.
And you can't do the `var/body = loc` either because the ghost is being `forceMove`d to the `body`'s turf during the `New()`